ingester.py

```diff
@@ -280,35 +268,44 @@ def process_historical_hos(self, processed_dir, processed_file_details,  make_hi
                 with open(os.path.join(processed_dir, fname), newline='') as csvfile:
                     reader = csv.DictReader(csvfile)
                     for row in reader:
-                        # add our rows
-                        hist_csv_row = {}
-                        new_row ={}
-                        row["Source Data Timestamp"] = f["source_datetime"].isoformat()
-                        row["Processed At"] = processed_time
-                        row["Source Filename"] = f["filename"]
-
-                        header.update(row)
-
-                        # XXX is this a bug? What happens to headers not in the alias?
-                        # rename the headers based on alias
-                        for alias, name in new_col_names.items():
-                            if alias in row:
-                                new_row[name] = row[alias]
-                        ft = Feature(attributes=new_row)
-                        features.append(ft)
-                        hist_csv_rows.append(row)
+
+                        # clean the keys (column names) of the input row
+                        clean_row = {k.strip(): v for k, v in row.items()}
+                        clean_row["Source Data Timestamp"] = f["source_datetime"].isoformat()
+                        clean_row["Processed At"] = processed_time
+                        clean_row["Source Filename"] = f["filename"]
+
+                        # now iterate the rows and map the headers to field names
+                        # in the ArcGIS Online layer. The matching is done against
+                        # the AGOL field aliases, and basic sanitation is performed.
+                        out_row = {}
+                        for alias, name in agol_fields.items():
+                            if alias == "ObjectId":
+                                continue
+                            if alias.strip() in clean_row:
+                                out_row[name] = clean_row[alias.strip()]
+                            else:
+                                missing_aliases.append(alias)
+
+                        features.append(Feature(attributes=out_row))
+                        hist_csv_rows.append(clean_row)
             else:
                 print(f"{fname} has a filesize of {size}, not processing.")
 
+        if len(missing_aliases) > 0:
+            print("These CSV field names do not match ArcGIS Online aliases:")
+            for mia in set(missing_aliases):
+                print(f" - {mia}   (length: {len(mia)})")
+
         # historical for generating a new source CSV
+        # I don't understand what this is for, but it should work as it did -AC
```

Comment:
This is used in the case where we want to generate an entire new CSV to upload, rather than doing an overwrite. Cases where this matters is when there are new columns, for example - overwrite() doesn't add new columns.

Reply:
Ah, ok, so that's how we have the v3, v4, etc. csvs.